apps/govquests-frontend/src/domains/questing/components/QuestButton.tsx

```diff
@@ -0,0 +1,57 @@
+import Button from "@/components/ui/Button";
+import SignInButton from "@/domains/authentication/components/SignInButton";
+import { cn } from "@/lib/utils";
+import type React from "react";
+import { useMemo } from "react";
+
+interface QuestButtonProps {
+  status: "unstarted" | "started" | "completed";
+  isSignedIn: boolean;
+  onClaim: () => void;
+}
+
+const QuestButton: React.FC<QuestButtonProps> = ({
+  status,
+  isSignedIn,
+  onClaim,
+}) => {
+  const { text, action, disabled } = useMemo(() => {
+    switch (status) {
+      case "completed":
+        return {
+          text: "Claim Rewards",
+          action: onClaim,
+          disabled: false,
+        };
+      default:
+        return {
+          text: "Read the texts to claim",
+          action: () => {},
+          disabled: true,
+        };
+    }
+  }, [status, onClaim]);
+
+  if (!isSignedIn) {
+    return (
+      <SignInButton className="self-center mt-6 py-3 px-24 bg-secondaryHover text-white hover:bg-secondaryDisabled font-medium rounded-lg" />
+    );
```

Comment:
bg-secondaryHover and bg-secondaryDisabled seems to be switched, shouldn't be as following?

className="self-center mt-6 py-3 px-24 bg-secondaryDisabled text-white hover:bg-secondaryHover font-medium rounded-lg"

Reply:
You're right, I was too lazy at the moment to create the right names 😄  I'll fix it on my next PR